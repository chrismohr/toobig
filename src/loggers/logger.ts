export interface Result {
  readonly path: string;
  readonly size: number;
  readonly maxSize: number;
}

export interface Logger {
  log(result: Result): void;
  finalize(): Promise<void>;
}
