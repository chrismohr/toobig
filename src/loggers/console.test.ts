import ConsoleLogger from './console';
import { Logger } from './logger';

(global as any).console = {
  log: jest.fn(),
  debug: jest.fn(),
  info: jest.fn(),
  warn: jest.fn(),
  error: jest.fn(),
};

describe('console logger', () => {
  let logger: Logger;

  beforeEach(() => {
    logger = new ConsoleLogger();
  });

  describe('.log', () => {
    describe('for a failure result', () => {
      beforeEach(() => {
        logger.log({ path: 'foo', size: 100, maxSize: 80 });
      });

      it('writes the correct message to console.log', () => {
        // tslint:disable-next-line:no-console
        const calls = (console.log as jest.Mock).mock.calls;
        expect(calls).toMatchSnapshot();
      });
    });

    describe('for a success result', () => {
      beforeEach(() => {
        logger.log({ path: 'foo', size: 80, maxSize: 100 });
      });

      it('writes the correct message to console.log', () => {
        // tslint:disable-next-line:no-console
        const calls = (console.log as jest.Mock).mock.calls;
        expect(calls).toMatchSnapshot();
      });
    });
  });

  describe('.finalize', () => {
    it('returns a resolving promise', async () => {
      await expect(logger.finalize()).resolves.toBeUndefined();
    });
  });
});
