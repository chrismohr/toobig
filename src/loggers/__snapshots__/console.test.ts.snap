// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`console logger .log for a failure result writes the correct message to console.log 1`] = `
Array [
  Array [
    "[31mfoo 100B >= 80B[39m",
  ],
]
`;

exports[`console logger .log for a success result writes the correct message to console.log 1`] = `
Array [
  Array [
    "[31mfoo 100B >= 80B[39m",
  ],
  Array [
    "[32mfoo 80B < 100B[39m",
  ],
]
`;
